### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Zelinskyi-Serhii**

### CODE REVIEW

File: src/robotFactory.js [hunk 1]
```diff
@@ -1,10 +1,100 @@
...
+
+    if (cargo !== null) {
+      this.currentLoad.weight = cargo.weight;
+      this.currentLoad.description = cargo.description;
+    }
+
+    if (cargo === null || cargo.weight > this.maxLoadWeight) {
+      this.currentLoad = null;
+    }
+  }
```
Comment (gretoq) [inline src/robotFactory.js:92]: Too complicated solution for adding currentLoad where we have only two conditions

File: src/robotFactory.js [hunk 2]
```diff
@@ -1,10 +1,100 @@
...
+    };
+  }
+
+  getInfo() {
+    const { name, chipVersion, weight } = this;
+
+    return `Robot: ${name}, Chip version: ${chipVersion}, Weight: ${weight}`;
+  }
+
+  goForward(x = 1) {
```
Comment (gretoq) [inline src/robotFactory.js:23]: Complicated understanding your code. 'x' is bad name for variables, especially where you have 'this.coords.y' and you add 'x' to it.

File: src/robotFactory.js [hunk 3]
```diff
@@ -1,10 +1,100 @@
...
 
-class DeliveryDrone {}
+    this.coords = {
+      x: x,
+      y: y,
+    };
+  }
+
+  getInfo() {
+    const { name, chipVersion, weight } = this;
```
Comment (gretoq) [inline src/robotFactory.js:18]: You can delete this row of code and add 'this' to 'name', 'chipVersion' and 'weight'

**Review by gretoq — CHANGES_REQUESTED** [review]
Your implementation for 'hookLoad' is too complicated for understanding.

File: src/robotFactory.js (hunk 3 above)
Comment (OleksiiBlezniuk) [reply src/robotFactory.js:18]: You may leave it as it is now, but it's strange to desctructure `this` object

File: src/robotFactory.js [hunk 4]
```diff
@@ -1,10 +1,100 @@
...
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
```
Comment (OleksiiBlezniuk) [inline src/robotFactory.js:42]: Btw, do `BaseRobot` constructor expect that there will be extra property in `coords` object?

File: src/robotFactory.js [hunk 5]
```diff
@@ -1,10 +1,100 @@
...
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(name, weight, coords, chipVersion, maxLoadWeight, currentLoad) {
+    super(name, weight, coords, chipVersion);
+
+    this.maxLoadWeight = maxLoadWeight;
+
+    if (currentLoad === undefined) {
+      this.currentLoad = null;
+    } else if (currentLoad === null) {
```
Comment (OleksiiBlezniuk) [inline src/robotFactory.js:66]: Just set `null` as default value to `currentLoad` param

File: src/robotFactory.js [hunk 6]
```diff
@@ -1,10 +1,100 @@
...
+    this.coords.z += step;
+  }
+
+  goDown(step = 1) {
+    this.coords.z -= step;
+  }
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(name, weight, coords, chipVersion, maxLoadWeight, currentLoad) {
```
Comment (OleksiiBlezniuk) [inline src/robotFactory.js:59]: This line is too long, let's move each param to new line

File: src/robotFactory.js [hunk 7]
```diff
@@ -1,10 +1,100 @@
...
+
+    this.maxLoadWeight = maxLoadWeight;
+
+    if (currentLoad === undefined) {
+      this.currentLoad = null;
+    } else if (currentLoad === null) {
+      this.currentLoad = {
+        weight: null,
+        description: null,
+      };
```
Comment (OleksiiBlezniuk) [inline src/robotFactory.js:70]: No, if there is no `currentLoad` passed - you need to set `null` to it, it should be either correct object or null

File: src/robotFactory.js [hunk 8]
```diff
@@ -1,10 +1,100 @@
...
+    } else {
+      this.currentLoad = {
+        weight: currentLoad.weight,
+        description: currentLoad.description,
+      };
```
Comment (OleksiiBlezniuk) [inline src/robotFactory.js:75]: Just add `currentLoad` object, no need to re-create each property

File: src/robotFactory.js [hunk 9]
```diff
@@ -1,10 +1,100 @@
...
+
+    if (cargo !== null) {
+      this.currentLoad.weight = cargo.weight;
+      this.currentLoad.description = cargo.description;
+    }
+
+    if (cargo === null || cargo.weight > this.maxLoadWeight) {
+      this.currentLoad = null;
+    }
+  }
```
Comment (OleksiiBlezniuk) [inline src/robotFactory.js:92]: Too complicated, please fix it
There are two options - if cargo is valid - set cargo to currentLoad, if not - do nothing

**Review by OleksiiBlezniuk — CHANGES_REQUESTED** [review]

**Review by sergii-nosachenko — APPROVED** [review]

**Pull request closed without merging**